### PR TITLE
Allow master and worker spark configurations in cluster config

### PR DESCRIPTION
**Pull request by tmckayus**
This change allows the name of a configmap to be specified for
the master and/or worker nodes in a spark cluster.  The configmap
will be mounted at a well-known location and the SPARK_CONF_DIR
env var will be set to that location. This lets users specify
spark configuration files for the nodes.

### CODE REVIEW

**Comment by tmckayus** [comment]
This is WIP

File: helpers/podtemplates/podtemplates.go [hunk 1]
```diff
@@ -45,3 +46,22 @@ func (pt *OPodTemplateSpec) Containers(cntnrs ...*containers.OContainer) *OPodTe
 	pt.Spec.Containers = kcntnrs
 	return pt
 }
+
+func (pt *OPodTemplateSpec) SetConfigMapVolume(configmap string) *OPodTemplateSpec {
+	fmt.Println("configmap is", configmap)
```
Comment (tmckayus) [inline helpers/podtemplates/podtemplates.go]: debug code, remove

File: handlers/clusters.go [hunk 2]
```diff
@@ -309,19 +336,14 @@ func CreateClusterResponse(params clusters.CreateClusterParams) middleware.Respo
 	const masterSrvMsg = "Unable to create spark master service endpoint"
 	const imageMsg = "Cannot determine name of spark image"
 	const respMsg = "Created cluster but failed to construct a response object"
+        var masterconfdir string
```
Comment (tmckayus) [inline handlers/clusters.go]: done

File: helpers/podtemplates/podtemplates.go (hunk 1 above)
Comment (tmckayus) [inline helpers/podtemplates/podtemplates.go]: done

**Comment by tmckayus** [comment]
needs unit tests for SparkMasterConfig and SparkWorkerConfig fields in named configs

**Comment by elmiko** [comment]
thanks for the changes Trevor!

**Pull request merged**